### PR TITLE
Aodzip New Calculation Redstone

### PR DESCRIPTION
**Pull request by aodzip**
RedstoneLit and Redstone Wire works very well
Working on Other Redstone Tools

### CODE REVIEW

**Comment by keithkfng** [comment]
Nice!

**Comment by robske110** [comment]
Please merge this... I can't wait to try this out!

**Comment by keithkfng** [comment]
@aodzip can I merge this?

**Comment by aodzip** [comment]
No don't merge 
If anyone want try , check the Aodzip-Redstone2 branch

**Comment by aodzip** [comment]
If you merge this now , the Redstone system in master will DOOM

**Comment by keithkfng** [comment]
XD

**Comment by robske110** [comment]
So @aodzip you want us to test that before you merge it?

**Comment by keithkfng** [comment]
Started a Jenkins job for this so you can test it out - http://jenkins.imagicalcorp.ml:8080/job/ImagicalMine-RedstoneRewrite/

**Comment by aodzip** [comment]
yes

**Comment by aodzip** [comment]
Not just test bug 
All the redstone related block and item will rewrite
Now I just done some job on RedstoneTorch and RedstoneWire

**Comment by thebigsmileXD** [comment]
Hey! And i thought you are inactive!
I'll check out the changes and maybe change some inputs on that. I love how less pressure plates lag, maybe i'll fix buttons today. No idea why daylight detectors still don't do their job.

**Comment by aodzip** [comment]
@thebigsmileXD  seems You didn't find my branch before lol

File: src/pocketmine/level/Level.php [hunk 1]
```diff
@@ -133,6 +133,13 @@ class Level implements ChunkManager, Metadatable{
 	const BLOCK_UPDATE_SCHEDULED = 3;
 	const BLOCK_UPDATE_WEAK = 4;
 	const BLOCK_UPDATE_TOUCH = 5;
+	
+	const REDSTONE_UPDATE_PLACE = 1;
+	const REDSTONE_UPDATE_NORMAL_UNSTABLE = 2;
+	const REDSTONE_UPDATE_NORMAL = 3;
+	const REDSTONE_UPDATE_LOSTPOWER_UNSTABLE = 4;
+	const REDSTONE_UPDATE_LOSTPOWER = 5;
+	const REDSTONE_UPDATE_BREAK = 6;
```
Comment (thebigsmileXD) [inline src/pocketmine/level/Level.php]: Wouldn't this mess up?
Because constant for scheduled === constant for redstone normal?

Comment (aodzip) [inline src/pocketmine/level/Level.php]: Won't 
RedstoneBroadcast and onRedstoneUpdate is stabdalone

**Comment by thebigsmileXD** [comment]
I did, but i read "maybe i'll give up thiS branch"

**Comment by aodzip** [comment]
That is Aodzip-Redstone 
not Redstone2
lol

**Comment by aodzip** [comment]
I make 2 different method

**Comment by aodzip** [comment]
photo_2015-12-19_19-48-18

**Comment by aodzip** [comment]
@thebigsmileXD 
Try this code !

A huge Redstone Network and Redstone Tools update include redstone loop and multi powersource

TPS keep in 20
and load around 90%
CPU usage about 1%

**Comment by aodzip** [comment]
Who mess up matser branch ... 

default

**Comment by keithkfng** [comment]
No, there were some errors in your branch. Click on details to check out the errors.

**Comment by aodzip** [comment]
The Player.php
a commit here cause that 
https://github.com/ImagicalCorp/ImagicalMine/commit/1f286cf6feec9e94a605a52aee2ec64820eaefe2

**Comment by thebigsmileXD** [comment]
Did my latest commit with the power sources work?
I never pushed on branches before

**Comment by robske110** [comment]
I just can't wait to see this in the master branch ;)

**Comment by thebigsmileXD** [comment]
And yea, the performance is INCREDIBLE!
❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️❤️ i like that

**Comment by robske110** [comment]
Yes and i just can't wait for the new software here and yesterday an new hardware arrived 3.9Ghz @4Cores,  in know that Pocketmine was never good at multithreading but 4Ghz should be enough ;)

**Comment by thebigsmileXD** [comment]
Oh.. I committed to the wrong branch..

**Comment by aodzip** [comment]
@thebigsmileXD 
After add Redstone Tick , some call on onRedstoneUpdate will lost.
so some wire will work incorrect

**Comment by aodzip** [comment]
That branch should delete 
lol

**Comment by robske110** [comment]
When will this be meged?

**Comment by thebigsmileXD** [comment]
@aodzip could you add the things i committed to this branch? There was some cleanup and fixes for redstone inputs
And redstone torch toggles

**Comment by aodzip** [comment]
@thebigsmileXD 
Which commit and which branch?

**Comment by thebigsmileXD** [comment]
Latest commit in the Aodzip-Redstone branch

**Comment by keithkfng** [comment]
Finished exam? XD

**Comment by aodzip** [comment]
No , I just thougt a good code . I should write it down , otherwise i will forget it

**Comment by aodzip** [comment]
@thebigsmileXD 
Redstone Wire Calculation is finished.
Update redstone tools code to fit the new call
then , the pr is ready to merge

**Comment by robske110** [comment]
Yey
I love logic code those if statements remember me how fast those processors go over them just with a few transistors, and they just decide ON or OFF.
Realy like computers and programming, but I'm just starting to learn PHP, so I can not realy help you :)

**Comment by aodzip** [comment]
! M E R G E D !

**Comment by matcracker** [comment]
Very nice work!

**Comment by thebigsmileXD** [comment]
Already merged? Yay, i'm gonna change stuff now :p

**Comment by aodzip** [comment]
@thebigsmileXD
have you test the performance?
I think it is 100 times better the old one

**Comment by robske110** [comment]
WOOOOOOOOOOOOOOOOOOOOOOOW
I JUST THROWN MY SMARTPHONE AGAINST THE WALL, LUCKILY IT SURVIVED...
Realy like you @aodzip

**Comment by thebigsmileXD** [comment]
I LOVE IT!
THAT RUNS EVEN BETTER THAN IN PC SERVERS!

**Comment by matcracker** [comment]
If we update in php 7 it runs like god! *--*

**Comment by robske110** [comment]
If we update to PHP 384838838283784828 this runs like mydesk

**Comment by thebigsmileXD** [comment]
If we update to php 7 it wont work anymore.. So.. Running like my old pc

**Comment by aodzip** [comment]
@thebigsmileXD 
photo_2015-12-24_00-07-44

**Comment by aodzip** [comment]
ahhhhhhhhhhhhhhhhhhhhhhh
yayyyyyyyyyyyyyyyyyyyy
ohhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh
ahhahahahahaaaaaaaaaaaaaaaaa

**Comment by thebigsmileXD** [comment]
Omg, powered blocks.. HOW THE! AND RUNS BETTER THAN ON PC SERVERS? You can spam the lever.. 20 lamps.. No lag

**Pull request merged**